src/Xamarin.Android.Build.Tasks/Tasks/GetMicrosoftNuGetPackagesMap.cs

```diff
@@ -0,0 +1,111 @@
+#nullable enable
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Android.Build.Tasks;
+using System.Net.Http;
+using System.Diagnostics.CodeAnalysis;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Xamarin.Android.Tasks;
+
+public class GetMicrosoftNuGetPackagesMap : AndroidAsyncTask
+{
+	public override string TaskPrefix => "GNP";
+
+	/// <summary>
+	/// The cache directory to use for Maven artifacts.
+	/// </summary>
+	[Required]
+	public string MavenCacheDirectory { get; set; } = null!; // NRT enforced by [Required]
+
+	[Output]
+	public string? ResolvedPackageMap { get; set; }
+
+	public override async System.Threading.Tasks.Task RunTaskAsync ()
+	{
+		Directory.CreateDirectory (MavenCacheDirectory);
+
+		// We're going to store the resolved package map in the cache directory as
+		// "microsoft-packages-{YYYYMMDD}.json". If the file is older than today,
+		// we'll try to download a new one.
+		var all_files = PackagesFile.FindAll (MavenCacheDirectory);
+
+		if (!all_files.Any (x => x.IsToday)) {
+			// No file for today, download a new one
+			try {
+				using var http = new HttpClient ();
```

Comment:
`HttpClient` instances should be in `static` fields; see also: 

  * https://github.com/xamarin/java.interop/pull/1179#discussion_r1488461213
  * https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines#recommended-use
  * https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-net-http-httpclient#instancing